Lib/test/test_csv.py

```diff
@@ -736,6 +736,36 @@ def test_write_field_not_in_field_names_ignore(self):
         csv.DictWriter.writerow(writer, dictrow)
         self.assertEqual(fileobj.getvalue(), "1,2\r\n")
 
+    def test_dict_reader_accepts_only_sequences(self):
+        fieldnames = ["a", "b", "c"]
+        f = StringIO()
+        self.assertRaises(TypeError, csv.DictReader, f, iter(fieldnames))
+
+        try:
```

Comment:
There is no need for try-except here. If an exception is raised the test will fail. Ditto the other two tests.